DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp

```diff
@@ -409,3 +436,29 @@ std::string ArrowCsvForeignStorage::getType() const {
                "(storage_type='CSV:path/to/file.csv');`\n";
   return "CSV";
 }
+
+template <typename T, typename builderType>
+int64_t appendDecimalData(const std::shared_ptr<arrow::Array>& chunk,
+                          int chunkIdx,
+                          bool empty,
+                          std::shared_ptr<arrow::Array>& arrayOut) {
+  std::shared_ptr<arrow::Decimal128Array> decimalArray =
+      std::static_pointer_cast<arrow::Decimal128Array>(chunk);
+  builderType indexBuilder;
+  indexBuilder.Reserve(decimalArray->length());
+
+  for (int j = 0; j < decimalArray->length(); ++j) {
+    if (decimalArray->IsNull(chunkIdx) || empty ||
+        decimalArray->null_count() == decimalArray->length()) {
+      indexBuilder.Append(inline_int_null_value<T>());
+    } else {
+      auto valPtr = decimalArray->GetValue(j);
+      arrow::Decimal128 valDec128(valPtr);
+      T val = static_cast<int64_t>(valDec128);  // arrow can cast only to int64_t
```

Comment:
I'm not familiar with how they have decimal128 implemented, but do we need to be concerned about handling a legitimate int128 if it comes along?

Reply:
@andrewseidl When column type is decimal/numeric arrow will represent his content as a decimalArray every element of which is decimal128(~16byte int) always. Now this is the only way to get values from arrow to omnisci. If OmniSci will be support int128 it will be useful.